File: app/src/main/java/com/e/prismetrictestapplication/utilClasses/RetrofitClient.java
package com.e.prismetrictestapplication.utilClasses;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static RetrofitClient retrofitClient;
    private static Retrofit retrofit;

    private OkHttpClient.Builder builder = new OkHttpClient.Builder();
    //private HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();

    public static Retrofit getRetrofitInstance() {

        Gson gson = new GsonBuilder()
                .setLenient()
                .create();

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient okHttpClient = new OkHttpClient().newBuilder()
                .connectTimeout(6, TimeUnit.MINUTES)
                .readTimeout(6, TimeUnit.MINUTES)
                .writeTimeout(6, TimeUnit.MINUTES)
                .addInterceptor(interceptor)
                .build();


        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(ApplicationConstants.INSTANCE.BASEURLLAB)
                    .client(okHttpClient)
                    .addConverterFactory(GsonConverterFactory.create(gson))
                    .build();
        }
        return retrofit;
    }

    public static synchronized RetrofitClient getInstance() {
        if (retrofitClient == null) {
            retrofitClient = new RetrofitClient();
        }
        return retrofitClient;
    }
}

File: app/src/main/java/com/e/prismetrictestapplication/utilClasses/RetrofitClient.kt
package com.e.prismetrictestapplication.utilClasses

import com.google.gson.GsonBuilder
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

class RetrofitClient {
    private val builder: OkHttpClient.Builder = OkHttpClient.Builder()

    companion object {
        private var retrofitClient: RetrofitClient? = null
        private var retrofit: Retrofit? = null

        //private HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        val retrofitInstance: Retrofit?
            get() {
                val gson = GsonBuilder()
                    .setLenient()
                    .create()
                val interceptor = HttpLoggingInterceptor()
                interceptor.setLevel(HttpLoggingInterceptor.Level.BODY)
                val okHttpClient = OkHttpClient().newBuilder()
                    .connectTimeout(6, TimeUnit.MINUTES)
                    .readTimeout(6, TimeUnit.MINUTES)
                    .writeTimeout(6, TimeUnit.MINUTES)
                    .addInterceptor(interceptor)
                    .build()
                if (retrofit == null) {
                    retrofit = Retrofit.Builder()
                        .baseUrl(ApplicationConstants.Companion.BASEURLLAB)
                        .client(okHttpClient)
                        .addConverterFactory(GsonConverterFactory.create(gson))
                        .build()
                }
                return retrofit
            }

        @get:Synchronized
        val instance: RetrofitClient?
            get() {
                if (retrofitClient == null) {
                    retrofitClient = RetrofitClient()
                }
                return retrofitClient
            }
    }
}